site/docs/09-math/07-vector.mdx

```diff
@@ -61,3 +61,169 @@ console.log(anotherPoint.toString()) // "(50, 50)"
+
+```typescript
+const scaled = vec(2, 3).scale(2); // (4, 6)
+const componentScaled = vec(2, 3).scale(vec(2, 0.5)); // (4, 1.5)
+```
+
+#### `normalize(): Vector`
+Returns a unit vector in the same direction.
+
+```typescript
+const unit = vec(3, 4).normalize(); // (0.6, 0.8)
+```
+
+### Distance and Magnitude
+
+#### `distance(v?: Vector): number`
+Returns the distance to another vector, or magnitude if no vector provided.
+
+```typescript
+const dist = vec(0, 0).distance(vec(3, 4)); // 5
+const magnitude = vec(3, 4).distance(); // 5
+```
+
+#### `squareDistance(v?: Vector): number`
+Returns the squared distance (faster than distance for comparisons).
+
+### Vector Math
+
+#### `dot(v: Vector): number`
+Calculates the dot product with another vector.
+
+```typescript
+const dotProduct = vec(1, 2).dot(vec(3, 4)); // 11
+```
+
+#### `cross(v: Vector | number): number | Vector`
+Calculates the cross product. With a vector, returns a scalar. With a scalar, returns a vector.
+
+```typescript
+const crossScalar = vec(1, 2).cross(vec(3, 4)); // -2
+const crossVector = vec(1, 2).cross(5); // (10, -5)
+```
+
+#### `perpendicular(): Vector`
+Returns a perpendicular vector (rotated 90 degrees).
+
+```typescript
+const perp = vec(1, 0).perpendicular(); // (0, -1)
+```
+
+#### `normal(): Vector`
+Returns the normalized perpendicular vector.
+
+### Rotation and Angles
+
+#### `toAngle(): number`
+Returns the angle of the vector in radians [0, 2π).
+
+```typescript
+const angle = vec(1, 1).toAngle(); // π/4 (45 degrees)
+```
+
+#### `rotate(angle: number, anchor?: Vector, dest?: Vector): Vector`
+Rotates the vector by an angle around an anchor point.
+
+```typescript
+const rotated = vec(1, 0).rotate(Math.PI / 2); // (0, 1)
+```
+
+#### `angleBetween(angle: number, rotationType: RotationType): number`
+Returns the angle needed to rotate this vector to match the target angle.
+
+### Utility Methods
+
+#### `equals(vector: Vector, tolerance?: number): boolean`
+Tests for equality within a tolerance.
+
+```typescript
+const isEqual = vec(1, 1).equals(vec(1.001, 0.999), 0.01); // true
+```
+
+#### `lerp(target: Vector, t: number): Vector`
+Linear interpolation between this vector and the target.  `t` is clamped between [0,1]
+
+```typescript
+const midpoint = vec(0, 0).lerp(vec(10, 10), 0.5); // (5, 5)
+```
+
+#### `average(vec: Vector): Vector`
+Returns the midpoint between this vector and another.
+
+#### `clone(dest?: Vector): Vector`
+Creates a copy of the vector.
+
+#### `negate(): Vector`
+Returns the negated vector.
+
+#### `clampMagnitude(magnitude: number): Vector`
+Clamps the vector's magnitude to a maximum value (mutates the vector).
+
+### Mutating Operations
+
+**⚠️ Warning**: These methods modify the original vector and should be used carefully.
+
+- `addEqual(v: Vector)`: Adds and mutates
+- `subEqual(v: Vector)`: Subtracts and mutates  
+- `scaleEqual(size: number)`: Scales and mutates
+- `setTo(x: number, y: number)`: Sets components and mutates
```

Comment:
@eonarheim any thoughts to my [proposal from way back here](https://github.com/excaliburjs/Excalibur/issues/3413#issuecomment-2845493717)? The fact this section is probably needed maybe points to needing a more intuitive convention.